homeassistant/components/homekit_controller/sensor.py

```diff
@@ -159,11 +159,85 @@ def state(self):
         return self._state
 
 
+class HomeKitBatterySensor(HomeKitEntity):
+    """Representation of a Homekit battery sensor."""
+
+    def __init__(self, *args):
+        """Initialise the entity."""
+        super().__init__(*args)
+        self._state = None
+        self._low_battery = False
+        self._charging = False
+
+    def get_characteristic_types(self):
+        """Define the homekit characteristics the entity is tracking."""
+        return [
+            CharacteristicsTypes.BATTERY_LEVEL,
+            CharacteristicsTypes.STATUS_LO_BATT,
+            CharacteristicsTypes.CHARGING_STATE,
+        ]
+
+    @property
+    def device_class(self) -> str:
+        """Return the device class of the sensor."""
+        return DEVICE_CLASS_BATTERY
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return "{} {}".format(super().name, "Battery")
```

Comment:
Please use f-strings instead of str.format.

Reply:
Done - migrated the other sensors in this module as well.